generator/src/main/java/com/scottlogic/deg/generator/decisiontree/MostProlificConstraintOptimiser.java

```diff
@@ -26,57 +25,36 @@
 import java.util.stream.Stream;
 
 public class MostProlificConstraintOptimiser implements DecisionTreeOptimiser {
-    private final int maxIterations;
-    private final int maxDepth;
-
-    public MostProlificConstraintOptimiser() {
-        this(50, 10000000);
-    }
-
-    public MostProlificConstraintOptimiser(int maxIterations, int maxDepth) {
-        this.maxIterations = maxIterations;
-        this.maxDepth = maxDepth;
-    }
+    private final int maxIterations = 50;
 
     @Override
     public DecisionTree optimiseTree(DecisionTree tree){
-        ConstraintNode newRootNode = optimiseLevelOfTree(tree.getRootNode(), 1);
-
-        if (newRootNode == null)
-            return tree;
-
+        ConstraintNode newRootNode = optimiseLevelOfTree(tree.getRootNode());
         return new DecisionTree(newRootNode, tree.getFields());
     }
 
-    private ConstraintNode optimiseLevelOfTree(ConstraintNode rootNode, int depth){
-        Collection<DecisionNode> decisions = rootNode.getDecisions();
-        if (decisions.size() <= 1 || depth > this.maxDepth)
-            return null; //not worth optimising
-
-        int iteration = 0;
-        int prevDecisionCount = decisions.size();
-        ConstraintNode newRootNode;
-        while (iteration < this.maxIterations && (newRootNode = optimiseDecisions(rootNode, depth)) != null)
-        {
-            rootNode = newRootNode;
+    private ConstraintNode optimiseLevelOfTree(ConstraintNode rootNode){
+        for (int iteration = 0; iteration < maxIterations; iteration++) {
+            ConstraintNode newRootNode = optimiseDecisions(rootNode);
 
-            int newDecisionCount = rootNode.getDecisions().size();
-            int changeInDecisionCount = newDecisionCount - prevDecisionCount;
-            if (Math.abs(changeInDecisionCount) < 1) {
-                break;
+            if (noChangeInDecisionCount(rootNode, newRootNode)) {
+                return newRootNode;
             }
 
-            prevDecisionCount = newDecisionCount;
-            iteration++;
+            rootNode = newRootNode;
         }
 
         return rootNode;
     }
 
-    private ConstraintNode optimiseDecisions(ConstraintNode rootNode, int depth){
+    private boolean noChangeInDecisionCount(ConstraintNode rootNode, ConstraintNode newRootNode) {
+        return newRootNode.getDecisions().size() - rootNode.getDecisions().size() == 0;
```

Comment:
`a - b == 0` is equivalent to `a == b` :)

Reply:
you didn't see anything